frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1874,6 +1931,28 @@ function ReaderHighlight:editHighlightStyle(page, i)
     self:showHighlightStyleDialog(apply_drawer, item.drawer)
 end
 
+function ReaderHighlight:editHighlightColor(page, i)
+    local item = self.view.highlight.saved[page][i]
+    local apply_color = function(color)
+        self:writePdfAnnotation("delete", page, item)
+        item.color = color
+        if self.ui.paging then
+            self:writePdfAnnotation("save", page, item)
```

Comment:
Do pdf annotations allow different colors?
Our implementation supports the drawer only so far
https://github.com/koreader/koreader/blob/39d54956ec42e97e0c3e57ea3ff994b6ebe57e31/frontend/document/pdfdocument.lua#L251-L258

Reply:
Yes, see https://github.com/koreader/koreader-base/blob/16a856a637421dc23956ab89925c46a113c5291a/ffi/mupdf.lua#L686.

Reply:
I guess the backend will need to be patched to use `item.color` instead of hardcoding colors.
```
    if type == M.PDF_ANNOT_HIGHLIGHT then
        color[0] = 1.0
        color[1] = 1.0
        color[2] = 0.0
```

However the frontend code to pass this in seems ok.